bin/cli.js

```diff
@@ -183,30 +183,67 @@ const handleDeprecatedEnvVariables = options => {
   ) {
     uiDeprecatedTag(
       i18n(`${i18nKey}.handleDeprecatedEnvVariables.portalEnvVarDeprecated`, {
-        configPath: getConfigPath(),
+        configPath: getConfigFilePath(),
       })
     );
     process.env.HUBSPOT_ACCOUNT_ID = process.env.HUBSPOT_PORTAL_ID;
   }
 };
 
+function addConfigEnvironmentVariablesMiddleware(options) {
+  const { useEnv, config } = options;
+
+  if (useEnv) {
+    process.env[ENVIRONMENT_VARIABLES.USE_ENVIRONMENT_HUBSPOT_CONFIG] = true;
+  }
+
+  if (config) {
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_CONFIG_PATH] = config;
+  }
+}
+
 /**
  * Auto-injects the derivedAccountId flag into all commands
  */
 const injectAccountIdMiddleware = async options => {
-  const { account } = options;
+  const { account: accountOption } = options;
 
   // Preserves the original --account flag for certain commands.
-  options.providedAccountId = account;
+  options.providedAccountId = accountOption;
 
-  if (options.useEnv && process.env.HUBSPOT_ACCOUNT_ID) {
-    options.derivedAccountId = parseInt(process.env.HUBSPOT_ACCOUNT_ID, 10);
-  } else {
-    options.derivedAccountId = getAccountId(account);
+  // Check environment variables for accountId
+  if (options.useEnv && process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ACCOUNT_ID]) {
+    options.derivedAccountId = parseInt(
+      process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ACCOUNT_ID],
+      10
+    );
+    return;
   }
+
+  // If not using environment variables, check if provided account exists
+  if (accountOption) {
+    const account = getConfigAccountIfExists(accountOption);
+    if (account) {
+      options.derivedAccountId = account.accountId;
+    } else {
+      logger.error(
+        i18n(`${i18nKey}.injectAccountIdMiddleware.accountNotFound`, {
+          accountIdentifier: accountOption,
+          authCommand: uiCommandReference('{authCommand}'),
+        })
+      );
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
+
+  // If no provided account or environment variable, use the default account
+  try {
```

Comment:
We only want to do this if `options.derivedAccountId` is undefined right?

Reply:
Oh yes good catch